core/drand_control.go

```diff
@@ -675,7 +698,36 @@ func (d *Drand) Status(c context.Context, in *drand.StatusRequest) (*drand.Statu
 		}
 	}
 
-	return &drand.StatusResponse{Dkg: &dkgStatus, Reshare: &reshareStatus, ChainStore: &chainStore, Beacon: &beaconStatus}, nil
+	// remote network connectivity
+	var resp = make(map[string]bool)
+	for _, addr := range in.GetCheckConn() {
+		if addr == d.priv.Public.Addr {
+			continue
+		}
+		// TODO check if TLS or not
```

Comment:
are these going to work with both tls and non tls peers?

Reply:
Currently no. I'm not sure how to handle this since in the address we dont have the TLS abbreviation, it's separate. 
What has been done so far is to use the `--insecure` flag to say all the address listed in the CLI arguments are to be treated as not TLS. But I can't differentiate _per address_ . Is this an acceptable solution at the moment you think ? I think so but I wanted to deal with that part last as I wasn't sure. I still need to make CI green on this one - working on it.

Reply:
I changed this to handle TLS and not tls now - but in all or nothing basis: either all addresses are TLS or none are. That's the best we can do now "quickly" but the messages I put in place should allow for different behaviors if we want it to.